view/frontend/web/js/getid.js

```diff
@@ -0,0 +1,30 @@
+define(['mage/url'], function (url) {
+    return async function getid() {
+        let sfCuid = localStorage.getItem('sf_cuid') || null;
+    
+        if (! sfCuid) {
+            const response = await fetch(url.build('/salesfire/ajax/getid'), {
```

Comment:
I think this needs adjusting slightly as it's going to cause a little issue.

If you add a .then(async response => await response.json()) and change const response to sfCuid = you then remove the other response.json() call below and add a return null into the catch.

The reason right now is that if this endpoint errors and returns something the response.json below can read it will set it to that, which may be an invalid value and not the cookie we're hoping for